src/network/networkd-can.c

```diff
@@ -119,6 +119,40 @@ static int link_set_can(Link *link) {
                         return log_link_error_errno(link, r, "Could not append IFLA_CAN_BITTIMING attribute: %m");
         }
 
+        if (link->network->can_data_bitrate > 0 || link->network->can_data_sample_point > 0) {
+                struct can_bittiming bt = {
+                        .bitrate = link->network->can_data_bitrate,
+                        .sample_point = link->network->can_data_sample_point,
+                };
+
+                if (link->network->can_data_bitrate > UINT32_MAX) {
+                        log_link_error(link, "data bitrate (%" PRIu64 ") too big.", link->network->can_data_bitrate);
+                        return -ERANGE;
+                }
```

Comment:
Please introduce a function `can_section_verify` and call this in `networkd-network.c:network_verify` and drop the verification here.

```
if (link->network->can_data_bitrate > 0) {
...
}

Reply:
I addressed this with a `config_parser_can_bitrate` function.